File: ios/Classes/NetworkManager.swift
// ios/Classes/NetworkManager.swift

import Foundation

public class NetworkManager {
  public func fetchAdConfig(placementId: String, callback: @escaping (String?) -> Void) {
    // Simulate fetching JSON config
    DispatchQueue.global().asyncAfter(deadline: .now() + 0.5) {
      let dummyJson = """
      {
        "placementId": "\(placementId)",
        "adType": "interstitial",
        "creativeUrl": "https://example.com/ad.jpg"
      }
      """
      callback(dummyJson)
    }
  }
}

File: android/src/main/kotlin/com/example/ad_platform_sdk/NetworkManager.kt
package com.example.ad_platform_sdk

import android.util.Log

class NetworkManager {

    fun fetchAdConfig(placementId: String, callback: (configJson: String?) -> Unit) {
        // Mock: return dummy JSON config, or call your real server async
        Thread {
            Thread.sleep(500)
            val dummyJson = """
                {
                  "placementId": "$placementId",
                  "adType": "interstitial",
                  "creativeUrl": "https://example.com/ad.jpg"
                }
            """.trimIndent()
            Log.d("NetworkManager", "Fetched config for $placementId")
            callback(dummyJson)
        }.start()
    }
}
